DEVELOPMENT.md

```diff
@@ -0,0 +1,90 @@
+## Development Guidelines
+
+This doc is a development guidance for Kubeflow/common developers.
+
+## Git development cycle
+
+We use [git flow](https://guides.github.com/introduction/flow/) workflow. The steps are
+
+1. Fork `kubeflow/common` repo. 
+
+1. Clone your fork under GOPATH,  add the `kubeflow/common` repository as a remote.
+
+    ```bash
+    mkdir -p ${go env GOPATH}/src/github.com/kubeflow
+    cd ${go env GOPATH}/src/github.com/kubeflow
+    
+    git clone https://github.com/YOUR_USERNAME/common
+    cd common
+    git remote add upstream https://github.com/kubeflow/common
```

Comment:
We can probably skip these steps in the doc and point to a GitHub tutorial for fork/pull request instead so this doc can focus on kubeflow/common.

Reply:
Good point. Point user to git flow official doc now.